src/app/util/IcdMonitoringTable.h

```diff
@@ -0,0 +1,80 @@
+/**
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app-common/zap-generated/cluster-objects.h>
+#include <lib/core/CHIPConfig.h>
+#include <lib/core/CHIPPersistentStorageDelegate.h>
+#include <lib/core/DataModelTypes.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/PersistentData.h>
+#include <stddef.h>
+
+namespace chip {
+
+constexpr size_t kIcdMonitoringBufferSize = 1024;
+
+struct IcdMonitoringEntry : public PersistentData<kIcdMonitoringBufferSize>,
+                            chip::app::Clusters::IcdManagement::Structs::MonitoringRegistrationStruct::Type
+{
+    IcdMonitoringEntry(FabricIndex fabric = kUndefinedFabricIndex, NodeId nodeId = kUndefinedNodeId)
+    {
+        this->fabricIndex      = fabric;
+        this->checkInNodeID    = nodeId;
+        this->monitoredSubject = nodeId;
+    }
+    bool IsValid() { return this->checkInNodeID != kUndefinedNodeId && this->fabricIndex != kUndefinedFabricIndex; }
+
+    CHIP_ERROR UpdateKey(StorageKeyName & key) override;
+    CHIP_ERROR Serialize(TLV::TLVWriter & writer) const override;
+    CHIP_ERROR Deserialize(TLV::TLVReader & reader) override;
+    void Clear() override;
+
+    uint16_t index = 0;
+};
+
+/**
+ * @brief IcdMonitoringTable exists to manage the persistence of entries in the ClientMonitoring Cluster.
+ *        To access persisted data with the IcdMonitoringTable class, instantiate an instance of this class
+ *        and call the LoadFromStorage function.
+ *
+ *        This class can only manage one fabric at a time. The flow is load a fabric, execute necessary operations,
+ *        save it if there are any changes and load another fabric.
+ *
+ *        Issue to refactor the class to use one entry for the entire table
+ *        https://github.com/project-chip/connectedhomeip/issues/24288
+ */
+
+struct IcdMonitoringTable
+{
+    IcdMonitoringTable(PersistentStorageDelegate * storage, FabricIndex fabric) : mStorage(storage), mFabric(fabric), mLimit(1)
+    {
+        chip::app::Clusters::IcdManagement::Attributes::ClientsSupportedPerFabric::Get(0, &mLimit);
```

Comment:
should do a check that endpoint 0 contains the `IcdManagement` cluster and the `ClientsSupportedPerFabric` attribute. Or is it a certitude that if we end up using IcdMonitoringTable, the cluster and attributes is there

Reply:
This table is used only by the IcdMonitoring server. If the cluster is not included, this class is never instantiated.